tools/spectral/ipa/rulesets/IPA-109.yaml

```
@@ -0,0 +1,15 @@
+# IPA-109: Custom Methods
+# http://go/ipa/109
+
+functions:
+  - eachCustomMethodMustBeGetOrPost
+
+rules:
+  xgen-IPA-109-custom-method-must-be-GET-or-POST:
+    description: 'The HTTP method for custom methods must be GET or POST. http://go/ipa/109'
+    message: '{{error}} http://go/ipa/109'
+    severity: warn
+    given: '$.paths'
+    then:
+      field: '@key'
+      function: 'eachCustomMethodMustBeGetOrPost'
```

```suggestion
    given: '$.paths'
    then:
      function: 'eachCustomMethodMustBeGetOrPost'
```
Since we are evaluating the path items in this case, you can pass them directly here, and then in the function you don't need to retrieve them from the documentInventory